ios/ReactNativePageView.m

```diff
@@ -461,4 +466,28 @@ - (NSString *)determineScrollDirection:(UIScrollView *)scrollView {
 - (BOOL)isLtrLayout {
     return [_layoutDirection isEqualToString:@"ltr"];
 }
+
+
+- (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer {
+
+    // The below snippet disables the pager view's scrollview's scroll when current index is 0 and user is swiping back. Useful for fullScreenGestureEnabled in react-native-screens
+    if (gestureRecognizer == self.panGestureRecognizer) {
+        UIPanGestureRecognizer* panGestureRecognizer = (UIPanGestureRecognizer*) gestureRecognizer;
+        CGPoint velocity = [panGestureRecognizer velocityInView:self];
+        BOOL isLTR = [self isLtrLayout];
+        BOOL isBackGesture = (isLTR && velocity.x > 0) || (!isLTR && velocity.x < 0);
+        if (otherGestureRecognizer == self.scrollView.panGestureRecognizer && self.currentIndex == 0 && isBackGesture) {
+            self.scrollView.panGestureRecognizer.enabled = false;
+            return NO;
+        }
+        
+        self.scrollView.panGestureRecognizer.enabled = self.scrollEnabled;
+        return YES;
+    }
+    
+    // Resetting here to default in case we add more recognizer.
+    self.scrollView.panGestureRecognizer.enabled = self.scrollEnabled;
```

Comment:
This LOC will be unreachable as of now as we added only one recognizer. But still resetting the value for sanity.